src/org/parosproxy/paros/core/scanner/VariantJSONQuery.java

```diff
@@ -189,8 +190,11 @@ private void parseValue(String fieldName) {
         // Check if the value is a string        
         if (chr == QUOTATION_MARK) {
             int beginToken = sr.getPosition();
-            while ((chr = sr.read()) != QUOTATION_MARK) {
-                if (chr == -1) {
+            while (true) {
+                chr = sr.read();
+                if (chr == BACKSLASH) { sr.read(); }

```

Comment:
The statements should be in a new line (just a matter of consistency with the rest of the code).